### PR TITLE
Added demo and files for load testing using AAD authentication

### PR DESCRIPTION
**Pull request by ninallam**
# Change Description

> Please include a summary of the changes.

## Linked GitHub Issue

> Link to any related github issue number (e.g. #56).

Fixes # (issue)

## Checklist

Please check all options that are relevant.

- [ ] I have made all necessary updates to the documentation.
- [ ] I have made all necessary updates to the provisioning scripts (bicep templates, github workflows).
- [ ] This is not a breaking change.

### CODE REVIEW

**Comment by mithunshanbhag** [comment]
@ninallam: Just wanted to quickly check if this PR was time-critical (i.e. for Microsoft Build Conference etc).

**Comment by ninallam** [comment]
@mithunshanbhag Yes, this the Build demo we have done for the session.

File: loadtests/aad-load-testing/config.yaml [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+splitAllCSVs: False
+configurationFiles:
+- Samsung phone.jpg
+failureCriteria:
+- View product: percentage(error) > 10
+- Product landing page: p90(response_time_ms) > 3000
+- View cart: p90(response_time_ms) > 100
+secrets:
+- name: clientId
+  value: https://contosotradersloadtest.vault.azure.net/secrets/clientId
```
Comment (mithunshanbhag) [inline loadtests/aad-load-testing/config.yaml:15]: The suffixes might vary (per the `SUFFIX` github repository secret). We shouldn't assume that it'll always be `test`.

File: demo-scripts/azure-load-testing/aad-authentication.md [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+
+## Before You Begin
+
+Please execute the steps outlined in the [deployment instructions](../../docs/deployment-instructions.md) to provision the infrastructure in your own Azure subscription.
+
+## Setting up the AAD credentials
+
+1. In the Azure portal, you can navigate to the Azure Load Testing resource in the `contoso-traders-rg{SUFFIX}` resource group.
+
+1. Go to **Identity blade** and turn on System assigned Managed Identity.
```
Comment (mithunshanbhag) [inline demo-scripts/azure-load-testing/aad-authentication.md:17]: We can automate this step with bicep.

**Comment by mithunshanbhag** [comment]
From the JMX, I see that you're calling the `token` endpoint directly: `https://login.microsoftonline.com/${tenantId}/oauth2/token` 

But this will pass only if the user is already authenticated (via an existing OAuth flow like implict grant, authorization code grant or client credentials grant).....  

Login flows have to be initiated from the `authorize` endpoint: `https://login.microsoftonline.com/${tenantId}/oauth2/v2.0/authorize`

See: https://login.microsoftonline.com/common/.well-known/openid-configuration

image

**Comment by macel94** [comment]
any news on this?

**Pull request closed without merging**